samples/ask-akka-agent/src/main/java/akka/ask/indexer/application/RagIndexingWorkflow.java

```diff
@@ -181,12 +186,11 @@ private CompletionStage<Done> indexFile(Optional<Path> pathOpt) {
         logger.debug("Created {} segments for document {}", segments.size(), path.getFileName());
 
         return segments
-          .stream()
-          .reduce(
-            futDone,
-            (acc, seg) -> addSegment(seg),
-            (stage1, stage2) -> futDone
-          );
+            .stream()
+            .reduce(
+                futDone,
+                (acc, seg) -> addSegment(seg),
+                (stage1, stage2) -> futDone);
```

Comment:
Note that this is not correct, with this code it could just as well be a forEach and then `return futDon;` on a line after fire-and-forget adding all segments as asynchronous tasks.

I have made it entirely sequential in https://github.com/akka/akka-sdk/pull/363#discussion_r2043675856

We can revisit if it is worth trying to parallelize that or not, not sure it is worth the effort to handle separate segments of the same file in parallel, but maybe. Are the embed and store calls slow, I expect they go to open api and mongo db in cloud respectively?

Reply:
I think we should keep it simple and sequential. If we want to show a more advanced indexer as a later step it would probably involve @octonato distributed workers https://github.com/akka-samples/ask-akka-agent/pull/33

Reply:
Let's merge this first, and let @johanandren make the virtual thread changes afterwards.